pydicom/uid.py

```diff
@@ -59,18 +63,13 @@ def __new__(cls, val):
         pydicom.uid.UID
             The UID object.
         """
-        # Don't repeat if already a UID class then may get the name that
-        #   str(uid) gives rather than the dotted number
-        if isinstance(val, UID):
-            return val
-
         if isinstance(val, str):
-            return super(UID, cls).__new__(cls, val.strip())
+            return super().__new__(cls, val.strip())  # type: ignore
 
-        raise TypeError("UID must be a string")
+        raise TypeError("A UID must be created from a string")
 
     @property
-    def is_implicit_VR(self):
+    def is_implicit_VR(self) -> bool:
         """Return ``True`` if an implicit VR transfer syntax UID."""
```

Comment:
Nice - I think particularly the bool return values are a good early target for typing across all pydicom